slumber/server/http.py

```diff
@@ -109,8 +109,10 @@ def wrapper(request, *args, **kwargs):
             response = {'_meta': dict(status=403, message='Forbidden'),
                 'error': unicode(exception)}
         except ObjectDoesNotExist, exception:
-            response = {'_meta': dict(status=404, message='Not Found'),
-                'error': unicode(exception)}
+            response = {
+                '_meta': dict(status=404, message='Not Found'),
+                'error': unicode(exception),
+                'id': unicode(request.path.rsplit('/', 2)[-2])}
```

Comment:
This looks really quite fragile to me -- you don't actually know the format of the URL when you catch the exception. Operations can do this any way they like.

I think we have to catch the exception in the relevant operations (at least the `data` one for now) and turn it into a 404 by catching the exception and then setting the `_meta` there. We also can't assume that the primary key is called `id`.